doc/source/whatsnew/v1.1.1.rst

```diff
@@ -20,6 +20,7 @@ Fixed regressions
 - Fixed regression where :func:`pandas.testing.assert_series_equal` would raise an error when non-numeric dtypes were passed with ``check_exact=True`` (:issue:`35446`)
 - Fixed regression in :class:`pandas.core.groupby.RollingGroupby` where column selection was ignored (:issue:`35486`)
 - Fixed regression in :meth:`DataFrame.shift` with ``axis=1`` and heterogeneous dtypes (:issue:`35488`)
+- Fixed regression in :meth:`DataFrame.diff` with read-only data (:issue:`35559`).
```

Comment:
i think no period at the end, otherwise LGTM

Reply:
we are likely inconsistent about this generally, but fixed here.